### DISCUSSION TITLE
CommadLineArgs on Windows and WASM

### DISCUSSION THREAD

**Discussion by TopperDEL** [unoplatform/uno#16017] [Q&A] [answered]
I'm confused about the way Startup/Launch-Parameters are to be handled.

I want my app to get info about what to do via start-parameters. In a Browser-Situation, this would be something like "https://www.myapp.com/?myParam=123"

On the console, it would be "myapp.exe myparam=123".

Using the LaunchSettings.json, this would look like this:
![image](https://github.com/unoplatform/uno/assets/1833242/8e01b934-1d65-4494-ac5f-448c498bb866)

On WASM, I get the parameter in the first Navigation to the MainPage via "args.Rgument" (that got passed-in via OnLaunched). But on WinUI, this is empty. On WinUI I can get acces to that parameter by calling `var commandLineArgs = Environment.GetCommandLineArgs();`. But that does not seem to be right - shouldn't these also be provided in `args.Arguments`?

Am I doing something wrong?

**Comment by jeromelaban** [comment, marked as answer]
@TopperDEL this is a behavior difference that we cannot handle on our end. WinAppSDK does its own behavior and it's not reporting CLI parameters through `args.Arguments`. It may be a good idea to report it in the WinAppSDK repository, though.

**Reply by TopperDEL** [reply]
Ah, thanks! I had the assumption that I was missing something here.

**Reply by TopperDEL** [reply]
Started a discussion in WindowsAppSdk-Repo: https://github.com/microsoft/WindowsAppSDK/discussions/4287
